monai/data/dataset.py

```diff
@@ -289,3 +290,99 @@ def __getitem__(self, index):
+    Dataset for segmentation and classification tasks based on array format input data and transforms.
+    It can apply same random operations for both image transforms and segmentation label transforms.
+    The `transform` can be :py:class:`monai.transforms.Compose` or any other callable object.
+    For example:
+    If train based on Nifti format images without metadata, all transforms can be composed::
+
+        img_transform = Compose(
+            [
+                LoadNifti(image_only=True),
+                AddChannel(),
+                RandAdjustContrast()
+            ]
+        )
+
+    If train based on Nifti format images and the metadata, the array transforms can not be composed
+    because several transforms receives multiple parameters or return multiple values. Then Users need
+    to define their own callable method to parse metadata from `LoadNifti` or set `affine` matrix
+    to `Spacing` transform::
+
+        class TestCompose(Compose):
+            def __call__(self, input_):
+                img, metadata = self.transforms[0](input_)
+                img = self.transforms[1](img)
+                img, _, _ = self.transforms[2](img, metadata["affine"])
+                return self.transforms[3](img), metadata
+
+        img_transform = TestCompose(
+            [
+                LoadNifti(image_only=False),
+                AddChannel(),
+                Spacing(pixdim=(1.5, 1.5, 3.0)),
+                RandAdjustContrast()
+            ]
+        )
+
+    Recommend to use dictionary Datasets for complicated data pre-processing.
+
+    """
+
+    def __init__(self, img_files, seg_files=None, labels=None, img_transform=None, seg_transform=None):
+        """
+        Initializes the dataset with the filename lists. The transform `img_transform` is applied
+        to the images and `seg_transform` to the segmentations.
+
+        Args:
+            img_files (list of str): list of image filenames
+            seg_files (list of str): if in segmentation task, list of segmentation label filenames
+            labels (list or array): if in classification task, list of classification labels
+            img_transform (Callable, optional): transform to apply to image arrays
+            seg_transform (Callable, optional): transform to apply to segmentation label arrays
+
+        """
+        if seg_files is not None and len(img_files) != len(seg_files):
+            raise ValueError("Must have same number of image and segmentation label files.")
+
+        self.img_files = img_files
+        self.seg_files = seg_files
+        self.labels = labels
+        self.img_transform = img_transform
+        self.seg_transform = seg_transform
+
+    def __len__(self):
+        return len(self.img_files)
+
+    def __getitem__(self, index):
+        img = self.img_files[index]
+        seg = self.seg_files[index] if self.seg_files is not None else None
+        label = self.labels[index] if self.labels is not None else None
```

Comment:
looks like there's no fundamental difference between the concept of `seg` and `label`. `label` might need some preprocessing as well, for example, converting the datatype from float to integer. 

so i think we could unify this class's members to have `self.input_sequence`, `self.target_sequence`, `self.input_transform`, `self.target_transform` to deal with input and target for supervised learning. if the sequences are filenames, users need to make sure that there is a file loader in the `input_transform` or `target_transform`